genie-web/src/main/java/com/netflix/genie/web/apis/rest/v3/controllers/JobRestController.java

```diff
@@ -946,10 +950,13 @@ private String agentExecution(
 
         if (attachments != null) {
             jobSubmissionBuilder.withAttachments(
-                Arrays
-                    .stream(attachments)
-                    .map(MultipartFile::getResource)
-                    .collect(Collectors.toSet())
+                this.attachmentService.saveAttachments(
```

Comment:
I think we discussed this offline but curious if you tried putting this in service tier instead and ran into a problem?

Reply:
Refactoring was complex, specifically updating the embedded codepath.
Since we are about to delete it, I placed this here, which is not completely wrong IMO.
Once embedded is deleted, we can push this down as you suggest.